src/auxiliary.jl

```diff
@@ -18,7 +18,8 @@ function resize_coeffs1!{T<:Number}(coeffs::Array{T,1}, order::Int)
     lencoef = length(coeffs)
     order ≤ lencoef-1 && return nothing
     resize!(coeffs, order+1)
-    coeffs[lencoef+1:end] .= zero(coeffs[1])
+    z = zero(coeffs[1])
+    @__dot__ coeffs[lencoef+1:order+1] = z
```

Comment:
It is not really needed, you are right, but it yields a tiny performance improvement (at a very low cost in memory). The change you propose (`coeffs[lencoef+1:end] .= 0`) could yield the same or even better performance; yet, for `Taylor1{Taylor1{T}}` cases, and other mixtures, using `0` instead of `zero(coeffs[1])` produces slightly different things with respect to the order of the first coefficient. While this in general should not be important, I can think of cases where it may have consequences in the sense that the answer depends on the order of the coefficient.

For the time being, I'll leave it as it is, though it is a fair point to check those cases in detail because it may yield another performance improvement.